store/src/java/com/zimbra/cs/imap/ImapCredentials.java

```diff
@@ -185,4 +186,17 @@ protected void hideFolder(ImapPath path) {
     protected boolean isFolderHidden(ImapPath path) {
         return mHiddenFolders == null ? false : mHiddenFolders.contains(path);
     }
+
+    public void logout() {
+        if(mStore != null) {
+            MailboxStore store = mStore.getMailboxStore();
+            if(store != null && store instanceof ZMailbox) {
+                try {
+                    ((ZMailbox)store).logout();
+                } catch (ZClientException e) {
+                    ZimbraLog.imap.error("ZMailbox failed to logout", e);
+                }
+            }
+        }
```

Comment:
Is it worth setting mStore to null?  Or is that not necessary because ImapCredentials won't be used much more?